bittensor/commands/wallets.py

```diff
@@ -1099,3 +1099,81 @@ def create_transfer_history_table(transfers):
     table.pad_edge = False
     table.width = None
     return table
+
+
+class WalletSignCommand:
+    """
+    Executes the ``sign`` command to sign a message with the provided wallet or wallet hotkey.
+
+    This command signs a message using the provided wallet.
+
+    Usage:
+        The command generates a signature for a given message using the provided wallet
+
+    Optional arguments:
+        --wallet.hotkey: A hotkey to use to sign the message instead of the wallet coldkey
+
+    Example usage::
+
+        btcli wallet sign --wallet.name default --message '{"something": "here", "timestamp": 1719908486}'
+        btcli wallet sign --wallet.name default --wallet.name hotkey --message '{"something": "here", "timestamp": 1719908486}'
+
+    Note:
+        When using `btcli`, `w` is used interchangeably with `wallet`. You may use either based on your preference for brevity or clarity.
+        This command is essential for users to easily prove their ownership over a coldkey or a hotkey.
+    """
+
+    @staticmethod
+    def run(cli):
+        r"""Sign a message using the provided wallet or hotkey."""
+        wallet = bittensor.wallet(config=cli.config)
+        keypair = wallet.coldkey
```

Comment:
for unlocking the coldkey, could you wrap 

```
        try:
            wallet.coldkey
        except bittensor.KeyFileError:
            bittensor.__console__.print(
                ":cross_mark: [red]Keyfile is corrupt, non-writable, non-readable or the password used to decrypt is invalid[/red]:[bold white]\n  [/bold white]"
            )
```

Reply:
Sure I'll wrap it.